### PR TITLE
remove ability to update allocated storage via strategy map

### PR DESCRIPTION
**Pull request by ciaranRoche**
add ability to update max allocated storage via strategy map

Currently, aws will autoscale as needed from allocated storage up to max allocated storage value. 
We need to remove the ability to allow cro to update the allocated storage value, as we want to rely 
on aws auto scaling. 
We do wish to allow cro the ability to update max allocated storage value

## Overview

Jira: https://issues.redhat.com/browse/INTLY-7231

## Verification

- Clone this branch
- Run `make cluster/prepare`
- Run `make cluster/seed/managed/postgres
- Run `make run`
- Update `"AllocatedStorage":50` in the create strategy for postgres in the cloud-resources-aws-strategies config map
- Update `"MaxAllocatedStorage":500` in the create strategy for postgres in the cloud-resources-aws-strategies config map
- Ensure allocated storage is not updated (ensure it is not added to pending modifications) 
- Ensure MaxAllocatedStorage is updated to 500 GiB

### CODE REVIEW

File: pkg/providers/aws/provider_postgres.go [hunk 1]
```diff
@@ -696,11 +696,6 @@ func verifyPendingModification(mi *rds.ModifyDBInstanceInput, pm *rds.PendingMod
 			pendingModifications = false
 		}
 	}
-	if mi.AllocatedStorage != nil && pm.AllocatedStorage != nil {
-		if *mi.AllocatedStorage == *pm.AllocatedStorage {
-			pendingModifications = false
-		}
-	}
```
Comment (aidenkeating) [inline pkg/providers/aws/provider_postgres.go:703]: Should we be replacing this with MaxAllocatedStorage also?

Comment (ciaranRoche) [reply pkg/providers/aws/provider_postgres.go:703]: No, this is to check for pending modifications, such that we do not continue calling modify instance until the next maintenance window hits. Id the modification is in a pending state. 

max allocation storage is modified instantly

Comment (aidenkeating) [reply pkg/providers/aws/provider_postgres.go:703]: Ok, I see that documented here https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Overview.DBInstance.Modifying.html under `Settings for your DB Instances`

**Comment by aidenkeating** [comment]
/approve

Verified against MW Collab

**Pull request merged**